frontend/src/pages/LogInPage/SignUpForm.tsx

```diff
@@ -1,103 +1,79 @@
-import React, { useState } from "react";
+import React, { useState } from 'react';
 import {
   Box,
   Button,
   Input,
   Stack,
   Heading,
   Container,
-} from "@chakra-ui/react";
-import { useNavigate } from "react-router-dom";
-import axios from "axios";
-import BackButton from "src/components/BackButton";
-import PORT from "src/utils/constant";
+  Text,
+} from '@chakra-ui/react';
+import { useNavigate } from 'react-router-dom';
+import axios from 'axios';
+import BackButton from 'src/components/BackButton';
 
 const SignupForm: React.FC = () => {
-  const [name, setName] = useState("");
-  const [username, setUsername] = useState("");
-  const [email, setEmail] = useState("");
-  const [password, setPassword] = useState("");
-  const [profilePicture, setProfilePicture] = useState("");
-  const [error, setError] = useState<string>("");
+  const [name, setName] = useState('');
+  const [username, setUsername] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [file, setFile] = useState<File | null>(null);
+  const [error, setError] = useState<string>('');
+  const [loading, setLoading] = useState(false);
   const navigate = useNavigate();
 
   const handleSubmit = async (e: React.FormEvent) => {
     e.preventDefault();
 
+    if (!file) {
+      setError('Veuillez sélectionner une photo de profil.');
+      return;
+    }
+
+    setLoading(true);
+    const formData = new FormData();
+    formData.append('name', name);
+    formData.append('username', username);
+    formData.append('email', email);
+    formData.append('password', password);
+    formData.append('profilePicture', file);
     try {
-      const response = await axios.post(`http://localhost:${PORT}/api/signup`, {
-        name,
-        username,
-        email,
-        password,
-        role: "user",
-        profilePicture,
-      });
+      const response = await axios.post(
+        'http://localhost:5001/api/auth/signup',
```

Comment:
Can you use the constant PORT instead 5001 ?
```suggestion
        'http://localhost:5001/api/auth/signup',
```